views/default/js/elgg/admin/upgrade.js

```diff
@@ -0,0 +1,225 @@
+
+define(function(require) {
+	var $ = require('jquery');
+	var elgg = require('elgg');
+
+	var upgrades = $('.elgg-upgrade');
+	var upgrade;
+	var className;
+	var progressbar;
+	var upgradeStartTime;
+	var timer;
+	var counter;
+	var percent;
+	var errorMessages = [];
+	var offset = 0;
+	var numSuccess;
+	var numError;
+	var numProcessed = 0;
+	var total;
+
+	/**
+	 *
+	 */
+	init = function() {
+		$('#upgrade-run').click(run);
+
+		upgrades.each(function(key, value) {
+			// Initialize progressbar
+			$(value).find('.elgg-progressbar').progressbar();
+		});
+	};
+
+	/**
+	 * Runs all individual upgrades one at a time
+	 *
+	 * @param {Object} e Event object
+	 */
+	var run = function(e) {
+		e.preventDefault();
+
+		// Replace button with spinner when upgrade starts
+		$('#upgrade-run').addClass('hidden');
+		$('#upgrade-spinner').removeClass('hidden');
```

Comment:
We have a spinner module in core now
On Apr 17, 2015 8:05 PM, "Juho Jaakkola" <notifications@github.com> wrote:

> In views/default/js/elgg/admin/upgrade.js
> <https://github.com/Elgg/Elgg/pull/8217#discussion_r28616401>:
>
> > +			// Initialize progressbar
> > +			$(value).find('.elgg-progressbar').progressbar();
> > +		});
> > +	};
> > +
> > +	/**
> > +	 * Runs all individual upgrades one at a time
> > +	 *
> > +	 * @param {Object} e Event object
> > +	 */
> > +	var run = function(e) {
> > +		e.preventDefault();
> > +
> > +		// Replace button with spinner when upgrade starts
> > +		$('#upgrade-run').addClass('hidden');
> > +		$('#upgrade-spinner').removeClass('hidden');
>
> Need to think about the layout a bit more. Currently there's no space for
> a spinner. Neither for error messages.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8217/files#r28616401>.
>